packages/eslint-plugin-gestalt/src/prefer-box-as-tag.test.js

```diff
@@ -0,0 +1,55 @@
+// @flow strict
+import { RuleTester } from 'eslint';
+import { readFileSync } from 'fs';
+import path from 'path';
+import rule, { SUPPORTED_HTML_TAGS } from './prefer-box-as-tag.js';
+
+const ruleTester = new RuleTester();
+
+const parserOptions = {
+  sourceType: 'module',
+  ecmaVersion: 6,
+  ecmaFeatures: {
+    jsx: true,
+  },
+};
+
+const validCode = readFileSync(
+  path.resolve(__dirname, './__fixtures__/prefer-box-as-tag/valid.js'),
+  'utf-8',
+);
+
+const invalidHTMLTagWithPropsInput = readFileSync(
+  path.resolve(
+    __dirname,
+    './__fixtures__/prefer-box-as-tag/invalid/invalid-HTML-tag-with-props-input.js',
+  ),
+  'utf-8',
+);
+
+const invalidHTMLTagWithPropsOutput = readFileSync(
+  path.resolve(
+    __dirname,
+    './__fixtures__/prefer-box-as-tag/invalid/invalid-HTML-tag-with-props-output.js',
+  ),
+  'utf-8',
+);
+
+const ERROR_MESSAGES = SUPPORTED_HTML_TAGS.map((tag) => ({
+  message: `Use <Box as="${tag}"></Box>.`,
+}));
+
+ruleTester.run('prefer-box-as-tag', rule, {
+  valid: [
+    {
+      code: validCode,
+      parserOptions,
+    },
+  ],
+  invalid: [[invalidHTMLTagWithPropsInput, invalidHTMLTagWithPropsOutput]].map((code) => ({
+    code: code[0],
```

Comment:
@AlbertCarreras might be easier to read as

```js
invalid: [[invalidHTMLTagWithPropsInput, invalidHTMLTagWithPropsOutput]].map(([input, output]) => ({
    code: input,
    parserOptions,
    output: output,
    errors: ERROR_MESSAGES,
  })),
```

Reply:
DAH, haha DONE